euphonic/cli/neutron_band_structure.py

```diff
@@ -52,6 +46,9 @@ def get_parser() -> argparse.ArgumentParser:
                         dest='gaussian_y',
                         help='Width of Gaussian broadening on energy axis in '
                         'ENERGY_UNITS. (No broadening if unspecified.)')
+    parser.add_argument('--lines', action='store_true',
+                        help='Plot a conventional phonon band structure '
+                             'instead of coherent neutron spectrum.')
```

Comment:
It pretty much makes `euphonic-dispersion` obsolete unless you specifically want to look at a precalculated q-point path. But that functionality was going to be moved to a CASTEP-specific repository anyway?

So maybe this script should just replace `euphonic-dispersion` and claim the name. Where are we with the standalone repo for CASTEP tools?

Would make sense to also replace `euphonic-dos` with something similar to this.

Reply:
I would be happy for it to replace `euphonic-dispersion` as long as it retains the ability to read from existing .phonon files,  and in that case the plain bandstructure should probably be the default with some other argument (`--neutron`?) for INS spectra.

I can update `euphonic-dos` to take force constants in a separate PR

Reply:
Hmm, then it would be ambiguous what it does if both force-constants _and_ a pre-calculated path are present.

Maybe a separate `euphonic-plot-qpointmodes` that can read either a `.phonon` file or a `qpointphononmodes.json` file?

Reply:
I'll probably attempt #105 once the plotting stuff is closer to being completed.

In terms of the CASTEP tools there has been absolutely no development done on those so we don't need to worry about it currently!

Reply:
Summary of discussion last night:

- move lines functionality into `euphonic-dispersion`, so that it can read force constants _or_ phonon mode data. This way the presentation options (broadening etc) will always make sense for the program/mode being used.
- rename `euphonic-neutron-band-structure` to something more generic like `euphonic-intensity-map` and allow it to plot by DOS weights as well as coherent INS.
- extend `euphonic-intensity-map` to also plot from phonon mode data
- in the CASTEP case, assume that the user wants to use modes if passing a `.phonon` file, and interpolate from force constants if passing a `.castep_bin` file.
- extend `euphonic-dos` to interpolate from force constants.

Then these three programs each produce distinct outputs, and each can take the same range of input formats (where appropriate). This seems user-friendly. Some supporting work to help with all this would be:
- centralise some file-dispatching logic that identifies formats from file names and/or structure data file content
- add metadata to euphonic-generated JSON to make distinguishing these files as easy as possible
- consider including a version tag to facilitate backwards-compatibility logic

Of these proposals, I think
- [ ] move lines functionality into `euphonic-dispersion` and
- [ ] rename `euphonic-neutron-band-structure` to something more generic like `euphonic-intensity-map`

are firmly required for this PR, and

- [ ]  allow it to plot by DOS weights as well as coherent INS

is required before this feature branch can be closed. Why? Because the DOS should be the default behaviour, for consistency with `euphonic-dos`, and something like `--weights=coherent` should be set to obtain the current output. That's a compatibility-breaking change.

Reply:
> * centralise some file-dispatching logic that identifies formats from file names and/or structure data file content

This could probably safely go in `cli.utils`

> * add metadata to euphonic-generated JSON to make distinguishing these files as easy as possible
> * consider including a version tag to facilitate backwards-compatibility logic

I can make a PR for these sooner rather than later if this would help

Reply:
Ok, I reworked the file-loading so intensity-map can also load precalculated phonon spectra! It should now be possible to load the same data files with `euphonic-dispersion` and `euphonic-intensity-map`, regardless of whether they are force constants or phonon band structures.

I think this is actually done now apart from user docs. Which can be a separate PR on the feature branch?